crates/executor/src/executor.rs

```diff
@@ -17,258 +19,423 @@ use revm::{
+            })
+            .collect::<Result<BTreeMap<H160, AccountInfoChangeSet>, _>>()?;
 
-            new_account.account_state = if account.storage_cleared {
-                new_account.storage.clear();
-                wipe_storage = true;
-                AccountState::StorageCleared
-            } else {
-                AccountState::Touched
-            };
+        // add drained ether to beneficiary.
+        let beneficiary = crate::eth_dao_fork::DAO_HARDFORK_BENEFICIARY;
 
-            // Insert storage.
-            let mut storage = BTreeMap::new();
+        let beneficiary_db_account =
+            db.load_account(beneficiary).map_err(|_| Error::ProviderError)?;
+        let old = to_reth_acc(&beneficiary_db_account.info);
+        beneficiary_db_account.info.balance += drained_balance;
+        let new = to_reth_acc(&beneficiary_db_account.info);
 
-            // insert storage into new db account.
-            new_account.storage.extend(account.storage.into_iter().map(|(key, value)| {
-                storage.insert(key, (value.original_value(), value.present_value()));
-                (key, value.present_value())
-            }));
+        let beneficiary_changeset = AccountInfoChangeSet::Changed { new, old };
 
-            // Insert into change.
-            change.insert(
-                address,
-                AccountChangeSet { account: account_info_changeset, storage, wipe_storage },
-            );
-        }
+        // insert changeset
+        changesets.insert(beneficiary, beneficiary_changeset);
+
+        Ok(changesets)
     }
-    (change, new_bytecodes)
 }
 
-/// After transaction is executed this structure contain
-/// transaction [Receipt] every change to state ([Account], Storage, [Bytecode])
-/// that this transaction made and its old values
-/// so that history account table can be updated.
-#[derive(Debug, Clone)]
-pub struct TransactionChangeSet {
-    /// Transaction receipt
-    pub receipt: Receipt,
-    /// State change that this transaction made on state.
-    pub changeset: BTreeMap<Address, AccountChangeSet>,
-    /// new bytecode created as result of transaction execution.
-    pub new_bytecodes: BTreeMap<H256, Bytecode>,
+impl<'a, DB> BlockExecutor<ExecutionResult> for Executor<'a, DB>
+where
+    DB: StateProvider,
+{
+    fn execute(
+        &mut self,
+        block: &Block,
+        senders: Option<Vec<Address>>,
+    ) -> Result<ExecutionResult, Error> {
+        let Block { header, body, ommers } = block;
+        let senders = self.recover_senders(body, senders)?;
+
+        self.init_block_env(header);
+
+        let mut cumulative_gas_used = 0;
+        // output of verification
+        let mut changesets = Vec::with_capacity(body.len());
+
+        for (transaction, sender) in body.iter().zip(senders.into_iter()) {
+            // The sum of the transaction’s gas limit, Tg, and the gas utilised in this block prior,
```

Comment:
```suggestion
            // The sum of the transaction’s gas limit, Tg, and the gas utilised in this block prior,
```

what's tg?

Reply:
I have no idea, maybe @rakita, as the original author and the owner, might answer for it.

Reply:
This is from yellow paper Tg is `transaction’s gas limit,`